tests/test_component.py

```diff
@@ -32,6 +32,7 @@ class Media:
 
         comp = SimpleComponent("simple_component")
         context = Context(comp.context(variable="test"))
+        comp.compile_instance_template({})
```

Comment:
As the code is currently set up, I think `compile_instance_template` has to be at the same level of external visibility as the constructor.  My thinking was that users really shouldn't be creating component instances on their own (i.e., other than through the template tags), and that this test is already violating encapsulation a bit by creating the component directly.

It would be possible to make this test pass without calling `compile_instance_template` by having the `render` function call it with no slots if it hasn't been called, but I think that risks hiding errors.

Reply:
Makes sense!